targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,353 @@
+#if !DISABLE_PLAYFABCLIENT_API
+
+using PlayFab.ClientModels;
+using PlayFab.Internal;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+
+namespace PlayFab.UUnit
+{
+    public class PluginManagerTest : UUnitTestCase
+    {
+        private class CustomSerializerPlugin : ISerializerPlugin
+        {
+            public bool IsSerializeCalled = false;
+
+            public T DeserializeObject<T>(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public T DeserializeObject<T>(string serialized, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object DeserializeObject(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+
+            public string SerializeObject(object obj)
+            {
+                this.IsSerializeCalled = true;
+                return "{}";
+            }
+
+            public string SerializeObject(object obj, object serializerStrategy)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class CustomTransportPlugin : ITransportPlugin
+        {
+            public string Name;
+            public bool IsDoPostCalled = false;
+
+            public async Task<object> DoPost(string urlPath, object request, Dictionary<string, string> headers)
+            {
+                this.IsDoPostCalled = true;
+                return "{}";
+            }
+        }
+
+        private class CustomApi
+        {
+            public string PluginName;
+
+            public async Task<object> SomeApiMethod()
+            {
+                var transport = (ITransportPlugin)PluginManager.GetPlugin(PluginContract.PlayFab_Transport, this.PluginName);
+                return await transport.DoPost(null, null, null);
```

Comment:
For this test, I wouldn't add the extra complication of using the "real" transport plugin, nor the async.
Both of those add complexity, but don't really improve the "test N plugin clones" case.
I would create two totally custom plugin types: CustomApi and CustomTransport, and not reuse ANY existing plugins, expectations, or interfaces.

For example, if we change the signature of DoPost, we have to change this test, when in reality, this test has no required connection to the DoPost function signature.